backend/src/backend/routes/netjson.clj

```diff
@@ -71,11 +71,52 @@
                                     (params "dateto")))
     :else (error-handler-rep 400 "Invalid query" req)))
 
+(defn adaption-check-if-exist
+  "Returns true if the adaption is registered, false otherwise"
+  [adaption_id]
+  (if (= [] (get-adaption-from-id adaption_id))
+    false
+    true))
+
+(defn device-check-if-exist
+  "Returns true if device is registered, false otherwise."
+  [device_id]
+  (if (= [] (get-device-from-id device_id))
+    false
+    true))
+
+(defn adaption-handler
+  "POST request to log adaption"
+  [{params :query-params :as req}]
+  (cond
+    (and
+     (contains? params "adaption_id")
+     (contains? params "device_id")
+     (contains? params "description")) (let [[adaption_id device_id description] [(read-string (params "adaption_id")) (read-string (params "device_id")) (params "description")]]
```

Comment:
let expressions can contain multiple bindings, you don't have to put them into a vector

Reply:
This goes for your other let bindings as well